instagram-app/src/components/PostContainer/PostPage.js

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
+//import withAuthenticate from './components/Authentication/withAuthenticate';
+import '../../App.css';
+import SearchBar from '../SearchBar/SearchBar';
+import dummyData from '../../dummy-data';
+import PostContainer from './PostContainer';
+
+class PostPage extends Component {
+  constructor() {
+    super();
+    this.state = {
+      data: [],
+      comments: {
+        text: '',
+        username: ''
+      }
+     
+    }
+  }
+  
+
+  addComment = (e) => {
+    e.preventDefault();
+    const thisIndex = e.target.getAttribute('index')
+    //console.log(this.state.data[thisIndex].comments)
+    this.state.data[thisIndex].comments.push({ text: this.state.comments.text, username: localStorage.getItem('user')})
+    this.setState({
+      data: [...this.state.data ]
+    })
+  }
+
+  handleChange = e => {
+    //console.log(e.target.value)
+    //this.state.data[0].comments.push({ text:e.target.value, username: "Billy Bob"})
+    this.setState({
+      data: [...this.state.data],
+      comments: {
+        text: e.target.value,
+        username: "BillyBob"
+      }
+    })
+  }
+
+ handleSearch = e => {
+    //console.log(e.target.value)
+    //this.setState({data: [...this.state.data]})
+    //const text = e.target.value.split("").toString()
+    //console.log(text)
+    //const newArray = this.state.data.filter(item => (this.setState({data: item})))
+    const newArray = []
+    this.state.data.filter(function(post) {
+      if (post.username === e.target.value) {
+       newArray.push(post)
+       console.log(newArray)
+       return newArray
+      } else {
+        return newArray
+      }
+      
+    });
+      
+        
+    if (newArray.length > 0) {
+      this.setState({data: newArray})
+    } else if (newArray.length === 0) {
+      this.setState({data: dummyData})
```

Comment:
If you change up the `.filter` statement starting at line 50/51 (see code comment above), there will be no need for this `if-else` statement; just use `this.setState({data: newArray}});` directly.